minari/data_collector/data_collector.py

```diff
@@ -316,11 +332,37 @@ def clear_buffer(dictionary_buffer: EpisodeBuffer, episode_group: h5py.Group):
             """
             for key, data in dictionary_buffer.items():
                 if isinstance(data, dict):
+
                     if key in episode_group:
                         eps_group_to_clear = episode_group[key]
                     else:
                         eps_group_to_clear = episode_group.create_group(key)
                     clear_buffer(data, eps_group_to_clear)
+                elif all([isinstance(entry, tuple) for entry in data]):
+                    # we have a list of tuples, so we need to act appropriately
+                    dict_data = {
+                        f"_index_{str(i)}": [entry[i] for entry in data]
+                        for i, _ in enumerate(data[0])
+                    }
+                    if key in episode_group:
+                        eps_group_to_clear = episode_group[key]
+                    else:
+                        eps_group_to_clear = episode_group.create_group(key)
+                    eps_group_to_clear.attrs["tuple"] = 1
+                    clear_buffer(dict_data, eps_group_to_clear)
+                elif all([isinstance(entry, OrderedDict) for entry in data]):
+
+                    # we have a list of OrderedDicts, so we need to act appropriately
+                    dict_data = {
+                        key: [entry[key] for entry in data]
+                        for key, value in data[0].items()
+                    }
+
+                    if key in episode_group:
+                        eps_group_to_clear = episode_group[key]
+                    else:
+                        eps_group_to_clear = episode_group.create_group(key)
+                    clear_buffer(dict_data, eps_group_to_clear)
                 else:
                     # convert data to numpy
                     np_data = np.asarray(data)
```

Comment:
Again, not your fault, but his function is very hard to read. We need to refactor it.